api/auth_middleware.go

```diff
@@ -37,30 +36,29 @@ func getServiceAccountID(ctx context.Context) (string, bool) {
 func authMiddleware(sasUC usecases.ServiceAccounts) func(http.Handler) http.Handler {
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			logger := middleware.GetLogger(r.Context())
 			header := r.Header.Get("authorization")
-			authHeaderContents := strings.Split(header, " ")
-			authHeader := buildAuth(header, authHeaderContents[0], authHeaderContents[1])
-			if !authHeader.isValid() {
-				w.WriteHeader(http.StatusUnauthorized)
+			authHeader, err := buildAuth(header)
+			if err != nil {
+				handleInvalidAuth(w, logger)
 				return
 			}
 
+			var errr error
```

Comment:
Drop this and keep using err? Reusing the err var is more usual than creating duplicate ones.